include/rdma/fabric.h

```diff
@@ -315,6 +315,28 @@ enum {
 	FI_PROTO_EFA
 };
 
+enum {
+	FI_TC_TYPE_UNSPEC,
+	FI_TC_TYPE_LABEL,
+	FI_TC_TYPE_DSCP,
+	FI_TC_MAX_TYPE,
+};
+
+#define FI_TC_BEST_EFFORT	(1ULL << 0)
+#define FI_TC_LOW_LATENCY	(1ULL << 1)
+#define FI_TC_DEDICATED_ACCESS	(1ULL << 2)
+#define FI_TC_BULK_IO_DATA	(1ULL << 3)
+#define FI_TC_SCAVENGER		(1ULL << 4)
+#define FI_TC_MANAGEMENT	(1ULL << 5)
+#define FI_TC_BULK_IPC_DATA	(1ULL << 6)
+#define FI_MAX_TC_DEFINED	FI_TC_BULK_IPC_DATA
+#define FI_TC_PROV_START	(1ULL << 31)
```

Comment:
Fair enough. I'll switch them to an enum, but I'm not picky about names. Suggest some names and we can just replace them.

Reply:
The names are hard -- let me look at the types of names DSCP uses and see if we can't come up with something usable.

Reply:
Based on tens of minutes of searching, these names are commonly used.  They match fairly close with the initial proposal, as much as I don't like the term 'scavenger'.  The only real change is management -> network control, which seems more commonly used in google search results.
```
enum {
    FI_TC_LABEL = 0x01000000,
    FI_TC_BEST_EFFORT,
    FI_TC_SCAVANGER,
    FI_TC_NETWORK_CTRL,
    FI_TC_BULK_DATA, /* high throughput */
    FI_TC_LOW_LATENCY, /* transactional */
    FI_TC_REALTIME_STREAM,
};
```
We can add others later as needed.  I don't know what dedicated access means, so I dropped that one.

Reply:
I've left dedicated access for now, but I've taken the other names. Dedicated access has a definition that is described on the man page and was discussed in the OFIWG meetings.

My understanding for that traffic class is that it should be used with applications that have VIP-like access to a fabric beyond the definition of best effort. It could be compared with FI_TC_NETWORK_CTRL, with additional constraints such as larger messages, latency and resiliency guarantees. I do not believe those same guarantees are provided with the management/network-control class.